### PR TITLE
Remove starting value in tween variable actions

### LINKED ISSUE #4270: Variable tweens are not setting the variable to starting value at creation.
When creating a variable tween, the starting value is not set as the variable value when creating the tween.
The variable is refreshed only after the next pass.

And this example quits instantly.
image

The only way to work around is to set the Variable as 255 before the Tween starts OR start the tween AFTER the variable checking 
image

Windows 10 with GDev version 5.0.140

### PR DESCRIPTION
**Pull request by AlexandreSi**
Fixes #4270

### CODE REVIEW

File: Extensions/TweenBehavior/JsExtension.js [hunk 1]
```diff
@@ -118,19 +118,46 @@ module.exports = {
...
       .addIncludeFile('Extensions/TweenBehavior/shifty_setup.js')
       .addIncludeFile('Extensions/TweenBehavior/tweentools.js')
       .setFunctionName('gdjs.evtTools.tween.tweenVariableNumber');
 
+    extension
+      .addAction(
+        'TweenSceneVariableNumber2',
+        _('Tween a number in a scene variable'),
+        _(
+          "Tweens a scene variable's numeric value from current value to another."
```
Comment (4ian) [inline Extensions/TweenBehavior/JsExtension.js:133]:
```suggestion
          "Tweens a scene variable's numeric value from its current value to another."
```
?

**Review by 4ian — APPROVED** [review]

File: Extensions/TweenBehavior/JsExtension.js [hunk 2]
```diff
@@ -674,9 +735,9 @@ module.exports = {
     behavior
       .addAction(
         'AddTextObjectCharacterSizeTween',
-        _('Add text size tween'),
+        _('Tween text size'),
         _(
-          'Add a tween animation for the text object character size (Note: the size can never be less than 1).'
+          'Tweens the text object character size from tis current value to a new one (Note: the size can never be less than 1).'
```
Comment (4ian) [inline Extensions/TweenBehavior/JsExtension.js:740]:
```suggestion
          'Tweens the text object character size from its current value to a new one (note: the size can never be less than 1).'
```

File: Extensions/TweenBehavior/JsExtension.js [hunk 3]
```diff
@@ -705,9 +766,9 @@ module.exports = {
     behavior
       .addAction(
         'AddObjectOpacityTween',
-        _('Add object opacity tween'),
+        _('Tween object opacity'),
         _(
-          'Add a tween animation for the object opacity (Value between 0 and 255).'
+          'Tweens the object opacity from its current value to a new one (Note: the value shall stay between 0 and 255).'
```
Comment (4ian) [inline Extensions/TweenBehavior/JsExtension.js:771]:
```suggestion
          'Tweens the object opacity from its current value to a new one (note: the value shall stay between 0 and 255).'
```

**Review by 4ian — APPROVED** [review]

File: Extensions/TweenBehavior/JsExtension.js [hunk 4]
```diff
@@ -179,7 +207,7 @@ module.exports = {
         _('Tween the camera rotation'),
         _('Tweens the camera rotation from the current angle to a new one.'),
         _(
-          'Tween the rotation of camera on layer _PARAM3_ to _PARAM2_ for _PARAM4_ms with easing _PARAM5_ as _PARAM1_'
+          'Tween the rotation of camera on layer _PARAM3_ to _PARAM2_ over _PARAM4_ms with easing _PARAM5_ as _PARAM1_'
```
Comment (ClementPasteau) [inline Extensions/TweenBehavior/JsExtension.js:210]: Just to confirm, does using `_PARAM4_ms` display "ms" in the editor?

Comment (AlexandreSi) [reply Extensions/TweenBehavior/JsExtension.js:210]: Yes!

File: Extensions/TweenBehavior/tweentools.ts [hunk 5]
```diff
@@ -158,6 +161,26 @@ namespace gdjs {
         getShiftyScene(runtimeScene).add(tween);
       };
 
+      export const tweenVariableNumber2 = (
```
Comment (ClementPasteau) [inline Extensions/TweenBehavior/tweentools.ts:164]: maybe add a small comment above this function explaining why we use this one now and not the previously deprecated one?

File: Extensions/TweenBehavior/tweenruntimebehavior.ts [hunk 6]
```diff
@@ -239,6 +240,39 @@ namespace gdjs {
...
+    /**
+     * Add an object variable tween.
+     * @param identifier Unique id to identify the tween
+     * @param variable The object variable to store the tweened value
+     * @param toValue End value
+     * @param easingValue Type of easing
+     * @param durationValue Duration in milliseconds
+     * @param destroyObjectWhenFinished Destroy this object when the tween ends
+     */
+    addVariableTween2(
```
Comment (ClementPasteau) [inline Extensions/TweenBehavior/tweenruntimebehavior.ts:252]: same here

**Review by ClementPasteau — APPROVED** [review]

File: Extensions/TweenBehavior/tweenruntimebehavior.ts (hunk 6 above)
Comment (4ian) [reply Extensions/TweenBehavior/tweenruntimebehavior.ts:252]: 👍

File: Extensions/TweenBehavior/tweentools.ts (hunk 5 above)
Comment (4ian) [reply Extensions/TweenBehavior/tweentools.ts:164]: 👍

Comment (AlexandreSi) [reply Extensions/TweenBehavior/tweentools.ts:170]: I added a detailed comment on the deprecated function

File: Extensions/TweenBehavior/tweenruntimebehavior.ts (hunk 6 above)
Comment (AlexandreSi) [reply Extensions/TweenBehavior/tweenruntimebehavior.ts:258]: I added a detailed comment on the deprecated function

**Pull request merged**